README.md

```diff
@@ -77,341 +106,424 @@ JSON representation of Patient document at RavenDB side
+This method retrieves the profilePicture for a given patient document:
 ```java
-try(CloseableAttachmentResult result= session.advanced().attachments().get(patient,names[0].getName())){
-	Attachment attachment = new Attachment();
-	attachment.setName(names[0].getName());
-	attachment.setMimeType(names[0].getContentType());
-	byte[] bytes = IOUtils.toByteArray(result.getData());
-	attachment.setBytes(bytes);
-	patientAttachment.setAttachment(attachment);
+public ProfilePicture getProfilePicture(Patient patient) {
+    ProfilePicture profilePicture = new ProfilePicture();
+
+    try (CloseableAttachmentResult result = session.advanced()
+            .attachments()
+            .get(patient.getId(), ATTACHMENT_NAME)) {
+
+
+        if (result == null)
+            return null;
+        InputStream data = result.getData();
+        byte[] bytes = IOUtils.toByteArray(data);
+        profilePicture.setBytes(bytes);
+        profilePicture.setName(ATTACHMENT_NAME);
+        
+        return profilePicture;
+    } catch (IOException e) {
+        logger.log(Level.SEVERE, "", e);
+    }
+
+    return null;
 }
 ```
+
 ## Queries
-RavenDB uses indexes to retrieve data but they don't work the same way as relational database indexes work. The main difference is that RavenDB's indexes are schema-less documented oriented. RavenDB requires an index to solve a query;that is why indexe is needed. The great think is that a programmer is not required to manually create indexes - RavenDB can deduct and create required index dynamically, by analyzing query at run time.
-All query samples that follow are based on dynamic indexes, generated by RavenDB's search engine. 
-The provided Patient type as the generic type parameter does not only define the type of returned results, but it also indicates that the queried collection will be Patients.
+In RavenDB, a query can only be satisfied by an index. You can create your own indexes, but if no appropriate index exists, 
+RavenDB will automatically create one on the fly. Indexes learn from each new query, and are continuously optimized to 
+satisfy all previous queries made to a given collection. In this way RavenDB is built to optimize the speed of reads with minimal 
+administration. (By default, indexes are updated asynchronously so that they don't impact write speed. In most use cases 
+this improves performance, but you can always have write operations wait for the index to update before returning using 
+`session.advanced().waitForIndexesAfterSaveChanges();`.)  
+
+Queries are created by chaining methods on `session.query()`, and executed by calling `.toList()`. Before being sent to the server, every
+query is translated to [RQL (RavenDB Query Language)](https://ravendb.net/docs/article-page/4.2/java/indexes/querying/what-is-rql) - 
+our SQL-like language. RQL was designed to expose RavenDB's query pipeline to the user. You can write your queries in RQL
```

Comment:
It's hard to read in markdown because of the link in the middle, but "our SQL-like language" belongs to the previous sentence, not the next one:
> ...every query is translated to [RQL (RavenDB Query Language)]() - our SQL-like language. RQL was designed...